server/src/main/java/io/spine/server/bus/Bus.java

```diff
@@ -99,4 +164,128 @@ public void unregister(D dispatcher) {
      * dispatchers of the bus.
      */
     protected abstract DispatcherRegistry<C, D> createRegistry();
+
+    /**
+     * Filters the given messages.
+     *
+     * <p>The implementations may apply some specific validation to the given messages.
+     *
+     * <p>If a message passes the filtering, it is included into the resulting {@link Iterable};
+     * otherwise, {@link StreamObserver#onNext StreamObserver.onNext} is called for that message.
+     *
+     * <p>The filtering may cause {@link IsSent} instances with {@code OK} status for those message
+     * that can be handled during the filtering itself. Such messages are technically valid, but
+     * should not be processed further. An example is a scheduled Command.
+     *
+     * @param messages the message to filter
+     * @param observer the observer to receive the negative outcome of the operation
+     * @return the message itself if it passes the filtering or
+     * {@link Optional#absent() Optional.absent()} otherwise
+     */
+    private Iterable<T> filter(Iterable<T> messages, StreamObserver<IsSent> observer) {
+        checkNotNull(messages);
+        checkNotNull(observer);
+        final Collection<T> result = new LinkedList<>();
+        for (T message : messages) {
+            final Optional<IsSent> response = preProcess(toEnvelope(message));
+            if (response.isPresent()) {
+                observer.onNext(response.get());
+            } else {
+                result.add(message);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Pre-processes the given message.
+     *
+     * <p>The pre-processing may include e.g. validation.
+     *
+     * <p>If the given message is completely processed and should not be posted via
```

Comment:
These two paragraphs aren't totally clear. Please explain (or rephrase) what you mean by "should be posted" or and "should not be posted".